### PR TITLE
Customized album song table

### PR DESCRIPTION
**Pull request by HinataIzayoi**
Add customized album song table for editing and translation editing.
Support editing on clicking.

### CODE REVIEW

File: src/components/AlbumCustomizedListing.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    const p = this.props;
+    this._isMounted = true;
+    if (!p.sourceSongName) return;
+
+    await Promise.all([
+      requestTranslationNS(p.lang0, p.sourceSongName),
+      requestTranslationNS(p.lang1, p.sourceSongName),
+      requestTranslationNS(p.lang2, p.sourceSongName),
+    ]).then((value) => {
+      if (this._isMounted) {
```
Comment (Kailang) [inline src/components/AlbumCustomizedListing.js:41]: Useless if.

File: src/components/AlbumCustomizedListing.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+  async onApplyTranslation() {
+    const p = this.props;
+    const s = this.state;
+    if (!p.name) {
+      this.setState({editing: false});
+      return;
+    }
+
+    await Promise.all([
+      p.app.translationUpdate({src: p.name, lang: p.lang0, namespace: 'name.artifact', text: s.albumName0}),
```
Comment (Kailang) [inline src/components/AlbumCustomizedListing.js:157]: `editingTranslation0 && p.app.translationUpdate({src: p.name, lang: p.lang0, namespace: 'name.artifact', text: s.albumName0}),`

File: src/components/AlbumCustomizedListing.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+export default class AlbumListing extends React.Component {
+  constructor(props) {
+    super(props);
+
+    /** @type {import('../App').default} */
+    this.app = props.app;
+
+    this.state = {
+      albums: [],
+      albumSystem: [],
```
Comment (Kailang) [inline src/components/AlbumCustomizedListing.js:222]: Remove.

File: src/components/AlbumCustomizedListing.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    const lang1 = e.target.value;
+    this.setState({lang1});
+  }
+
+  onLang2Change(e) {
+    const lang2 = e.target.value;
+    this.setState({lang2});
+  }
+
+  onLang0Change(e) {
```
Comment (Kailang) [inline src/components/AlbumCustomizedListing.js:245]: Move up.

File: src/components/AlbumCustomizedListing.js [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    return <div className="container-fluid">
+      <section className="Bgc($gray-700) P(30px) text-light shadow">
+        <h2 className="row Fw(n)">Songs</h2>
+      </section>
+      <Link to='/songs'><button className="btn btn-outline-secondary">System songs</button></Link>
+      <table className="table table-responsive">
+        <thead>
+          <tr>
+            <th className="W(15%)">Album</th>
+            <th className="W(30%)">Song</th>
```
Comment (Kailang) [inline src/components/AlbumCustomizedListing.js:275]: Remove width style.

File: src/translationService.js [hunk 6]
```diff
@@ -63,6 +71,16 @@ export async function requestTranslation(namespace, src) {
   return text;
 }
 
+export async function requestTranslationNS(lang, src) {
```
[the remark this answers is not in the document]
Comment (HinataIzayoi) [reply src/translationService.js:74]: Done.

File: src/translationService.js [hunk 7]
```diff
@@ -41,6 +41,14 @@ export function pushDict(key, text) {
   localStorage.setItem(TRANSLATION_DICT_KEY, JSON.stringify(dict_));
 }
 
+export function pushDictLang(key, text, lang) {
```
[the remark this answers is not in the document]
Comment (HinataIzayoi) [reply src/translationService.js:44]: Done.

File: src/App.js [hunk 8]
```diff
@@ -504,6 +507,14 @@ export default class App extends React.Component {
   }
 
   async translationUpdate({src, lang, namespace, text}) {
+    const key = `${lang}:${namespace}:${src}`;
```
[the remark this answers is not in the document]
Comment (HinataIzayoi) [reply src/App.js:510]: Done.

File: src/App.js [hunk 9]
```diff
@@ -58,6 +59,8 @@ const VERSION = 0;
 const INTENT = 'web';
 
 const DEVELOPMENT = 'development';
+const TRANSLATION_LANG_KEY = 'translationLang';
```
[the remark this answers is not in the document]
Comment (HinataIzayoi) [reply src/App.js:62]: Done.

File: src/components/AlbumCustomizedListing.js (hunk 1 above)
Comment (HinataIzayoi) [reply src/components/AlbumCustomizedListing.js:41]: Removed.

File: src/components/AlbumCustomizedListing.js (hunk 2 above)
Comment (HinataIzayoi) [reply src/components/AlbumCustomizedListing.js:157]: Done.

File: src/components/AlbumCustomizedListing.js (hunk 3 above)
Comment (HinataIzayoi) [reply src/components/AlbumCustomizedListing.js:222]: Removed.

File: src/components/AlbumCustomizedListing.js (hunk 4 above)
Comment (HinataIzayoi) [reply src/components/AlbumCustomizedListing.js:245]: Moved.

File: src/components/AlbumCustomizedListing.js (hunk 5 above)
Comment (HinataIzayoi) [reply src/components/AlbumCustomizedListing.js:275]: Done.

**Pull request merged**